syntax = "proto3";

package langchain;

service LangChain {
  rpc GenerateAnswer (UserQuery) returns (ChatbotAnswer);
  rpc SemanticSearch (UserQuery) returns (SemanticSearchResult);
  rpc RecommendPrompts (EmptyRequest) returns (RecommendedPrompts);
}

message UserQuery {
  string query_text = 1;  
}

message ChatbotAnswer {
  string answer_text = 1;
}

message SemanticSearchResult {
  repeated int64 notes_id = 1;
}

message EmptyRequest {}

message RecommendedPrompts {
  repeated string recommended_prompts = 1;
}
